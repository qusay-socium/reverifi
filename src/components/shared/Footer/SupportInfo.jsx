import React from 'react';
import { MapPin, Mail, Phone } from 'react-feather';
import { ReactComponent as Logo } from 'assets/logo.svg';
import {
  InfoGroup,
  ContactInfo,
  InfoGroupItem,
  EmailLink,
} from './footer.styles';

// eslint-disable-next-line react/function-component-definition
const SupportInfo = () => (
  <InfoGroup>
    <Logo />
    <ContactInfo>
      <MapPin />
      <InfoGroupItem to="/">714-427-1242</InfoGroupItem>
    </ContactInfo>
    <ContactInfo>
      <Mail />
      <EmailLink rel="noopener" href="mailto:support@example.com">
        support@example.com
      </EmailLink>
    </ContactInfo>
    <ContactInfo>
      <Phone />
      <InfoGroupItem to="/">319-764-9876</InfoGroupItem>
    </ContactInfo>
  </InfoGroup>
);

export default SupportInfo;
