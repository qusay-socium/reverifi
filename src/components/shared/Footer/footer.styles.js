import colors from 'styles/colors';
import mq from 'styles/media-query';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const SocialMediaIcons = styled.div`
  display: flex;
  background: ${colors.whiteSand};
  padding: 0.9375rem 0rem 0rem 0.9375rem;

  svg {
    margin: 1.438rem 0.625rem 0 0;
  }

  ${mq.tablet`
      justify-content: center;
  `}
`;

export const InfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background: ${colors.whiteSand};
  padding: 2.0625rem 0em 2.0625rem 1.0625rem;

  ${mq.tablet`
    flex-direction:row;
    flex-wrap: wrap;
    row-gap: 1.25rem;
    justify-content: space-evenly;
    padding-bottom: 3.0625rem;
    column-gap: 0.625rem;
  `}
`;

export const ContactInfo = styled.div`
  display: flex;
  column-gap: 0.625rem;
  align-items: center;
  color: ${colors.green};
`;

export const InfoGroup = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.875rem;
  padding-bottom: 1.8125rem;
`;

export const InfoGroupHeader = styled.p`
  color: ${colors.green};
  font-size: 1.125rem;
  font-weight: 700;
  margin-bottom: 1.875rem;
  width: max-content;
`;

export const InfoGroupItem = styled(Link)`
  font-size: 0.875rem;
  color: ${colors.gray};
  text-decoration: none;
  width: max-content;
`;

export const EmailLink = styled.a`
  font-size: 0.875rem;
  color: ${colors.gray};
  text-decoration: none;
`;

export const SubscribeWrapper = styled.div`
  color: ${colors.green};
`;

export const SubscribeInputWrapper = styled.div`
  height: 2.375rem;
  max-width: 21.25rem;
`;

export const FooterWrapper = styled.div`
  background-color: ${colors.mineShaft};
  color: ${colors.white};
  display: flex;
  align-items: center;
  justify-content: center;
  height: 4.125rem;

  p {
    font-size: 0.75rem;
  }
`;

export const Button = styled.button`
  padding: 0rem;
  border: none;
  cursor: pointer;
  color: ${colors.white};
  background-color: ${colors.green};
  height: 2.375rem;
  border-radius: 1.75rem;
  padding: 0 2rem;
`;
