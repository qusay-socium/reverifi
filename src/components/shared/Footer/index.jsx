import { ReactComponent as Facebook } from 'assets/images/facebook.svg';
import { ReactComponent as Instagram } from 'assets/images/instagram.svg';
import { ReactComponent as Linkedin } from 'assets/images/linkedin.svg';
import { ReactComponent as Youtube } from 'assets/images/youtube.svg';
import React from 'react';
import Button from '../Button';
import Input from '../Input';
import {
  FooterWrapper,
  InfoGroup,
  InfoGroupHeader,
  InfoGroupItem,
  InfoWrapper,
  SocialMediaIcons,
  SubscribeInputWrapper,
  SubscribeWrapper,
} from './footer.styles';
import SupportInfo from './SupportInfo';

/**
 * Home page footer section.
 *
 * @return {JSX.Element}
 */
function Footer() {
  return (
    <>
      <SocialMediaIcons>
        <Facebook />
        <Instagram />
        <Youtube />
        <Linkedin />
      </SocialMediaIcons>
      <InfoWrapper>
        <SupportInfo />
        <InfoGroup>
          <InfoGroupHeader>Helpful Links</InfoGroupHeader>
          <InfoGroupItem to="/">Sign in</InfoGroupItem>
          <InfoGroupItem to="/">Support</InfoGroupItem>
          <InfoGroupItem to="/">FAQs</InfoGroupItem>
        </InfoGroup>
        <InfoGroup>
          <InfoGroupHeader>Company</InfoGroupHeader>
          <InfoGroupItem to="/">About Us</InfoGroupItem>
          <InfoGroupItem to="/">Privacy</InfoGroupItem>
          <InfoGroupItem to="/">Term of use</InfoGroupItem>
          <InfoGroupItem to="/">Accessibility</InfoGroupItem>
        </InfoGroup>
        <SubscribeWrapper>
          <InfoGroupHeader>Subscribe to our newsletter</InfoGroupHeader>
          <SubscribeInputWrapper>
            <Input
              placeholder="Enter email address"
              rightElement={
                <Button onClick={() => {}} ariaLabel="Subscribe">
                  Subscribe
                </Button>
              }
            />
          </SubscribeInputWrapper>
        </SubscribeWrapper>
      </InfoWrapper>
      <FooterWrapper>
        <p>
          Copyright Â© {new Date().getFullYear()} reverifi. All rights reserved.
        </p>
      </FooterWrapper>
    </>
  );
}

export default Footer;
