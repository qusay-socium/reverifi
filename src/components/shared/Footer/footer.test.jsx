import { shallow } from 'enzyme';
import React from 'react';
import Footer from '.';

describe('<Footer />', () => {
  it('renders <Footer />', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
