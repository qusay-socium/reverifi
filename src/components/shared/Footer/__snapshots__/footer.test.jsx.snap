// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders <Footer /> 1`] = `ShallowWrapper {}`;
